### PR TITLE
GUID/UUID and Decimal/Numeric support

### PR DESCRIPTION
**Pull request by traceon**
- Added support for Decimal/Numeric values
- Added support for GUID/UUID values
- Fixed Nullable parameter handling
- Added Descriptor read/write ODBC API functions
- Reworked existing tests matrix
- Added new integration/unit tests for the new changes
- Some refactoring of type conversion/parsing code (+ added new for new types)
- Other fixes

### CODE REVIEW

File: driver/type_info.cpp [hunk 1]
```diff
@@ -2,6 +2,37 @@
...
+    {"Float64", TypeInfo {"DOUBLE", false, SQL_DOUBLE, 15, 8}},
+    {"Decimal", TypeInfo {"DECIMAL", false, SQL_DECIMAL, 1 + 2 + 38, 16}}, // -0.
+    {"UUID", TypeInfo {"GUID", false, SQL_GUID, 8 + 1 + 4 + 1 + 4 + 1 + 4 + 12, sizeof(SQLGUID)}},
+    {"String", TypeInfo {"TEXT", true, SQL_VARCHAR, TypeInfo::string_max_size, TypeInfo::string_max_size}},
+    {"FixedString", TypeInfo {"TEXT", true, SQL_VARCHAR, TypeInfo::string_max_size, TypeInfo::string_max_size}},
+    {"Date", TypeInfo {"DATE", true, SQL_TYPE_DATE, 10, 6}},
+    {"DateTime", TypeInfo {"TIMESTAMP", true, SQL_TYPE_TIMESTAMP, 19, 16}},
+    {"Array", TypeInfo {"TEXT", true, SQL_VARCHAR, TypeInfo::string_max_size, TypeInfo::string_max_size}},
+
+    {"LowCardinality(String)",
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/type_info.cpp]: Most definitely yes, however this is just a moved existing piece code. A separate PR could be dedicated to that.

File: driver/type_info.cpp [hunk 2]
```diff
@@ -2,6 +2,37 @@
...
+        TypeInfo {"TEXT", true, SQL_VARCHAR, TypeInfo::string_max_size, TypeInfo::string_max_size}}, // todo: remove
+    {"LowCardinality(FixedString)",
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/type_info.cpp]: This is an existing code, as well. Needs to be reworked, but probably better to do it separately.

File: driver/type_info.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (traceon) [reply driver/type_info.cpp]: There is a type_parser.cpp that dose some of that currently.

File: driver/type_info.cpp [hunk 3]
```diff
@@ -2,6 +2,37 @@
...
+            4}}, // With perl, python ODBC drivers INT is uint32 and it cant store values bigger than 2147483647: 2147483648 -> -2147483648 4294967295 -> -1
+    {"UInt32", TypeInfo {"INT", true, SQL_INTEGER, 10, 4}},
+    {"UInt64", TypeInfo {"BIGINT", true, SQL_BIGINT, 20, 8}},
+    {"Int8", TypeInfo {"TINYINT", false, SQL_TINYINT, 1 + 3, 1}}, // one char for sign
+    {"Int16", TypeInfo {"SMALLINT", false, SQL_SMALLINT, 1 + 5, 2}},
+    {"Int32", TypeInfo {"INT", false, SQL_INTEGER, 1 + 10, 4}},
+    {"Int64", TypeInfo {"BIGINT", false, SQL_BIGINT, 1 + 19, 8}},
+    {"Float32", TypeInfo {"REAL", false, SQL_REAL, 7, 4}},
+    {"Float64", TypeInfo {"DOUBLE", false, SQL_DOUBLE, 15, 8}},
+    {"Decimal", TypeInfo {"DECIMAL", false, SQL_DECIMAL, 1 + 2 + 38, 16}}, // -0.
```
[the remark this answers is not in the document]
Comment (traceon) [reply driver/type_info.cpp]: This is again an existing mechanism, and could be upgraded in a separate PR. The current behavior (this is used only when describing columns) should work. The `SQL_DECIMAL` (as they are reported here) are communicated as strings anyway.

File: driver/ut/statement_parameters_it.cpp [hunk 4]
```diff
@@ -0,0 +1,599 @@
...
+    , public ::testing::WithParamInterface<std::tuple<std::string, std::string>>
+{
+protected:
+    using DataType = T;
+};
+
+
+// TODO: GIUD/UUID tests are temporarily disabled until this worked around/fixed: https://github.com/ClickHouse/ClickHouse/issues/7463
+using DISABLED_ParameterColumnRoundTripGUIDSymmetric = ParameterColumnRoundTripSymmetric<SQLGUID>;
+TEST_P(DISABLED_ParameterColumnRoundTripGUIDSymmetric, Execute) { execute<DataType>(GetParam(), GetParam(), type_info_for("UUID"), false/* case_sensitive */); }
```
Comment (Enmk) [inline driver/ut/statement_parameters_it.cpp]: please make that multi-line

Comment (traceon) [reply driver/ut/statement_parameters_it.cpp]: Done.

**Pull request merged**